struct_env/struct_env.py

```diff
@@ -5,37 +5,82 @@ class Struct:
 
     def __init__(self, config=None):
         if config is None:
-            config = {"components": 2, "discount_reward": 1, "k_comp": None}
+            config = {"components": 2,
+                      "discount_reward": 1,
+                      "k_comp": None,
+                      "env_type": "uncorrelated"}
+        assert "components" in config and \
+               "discount_reward" in config and \
+               "k_comp" in config and \
+               "env_type" in config, \
+            "Missing env config"
+
         self.ncomp = config["components"]
         self.discount_reward = config["discount_reward"]
-        self.k_comp = self.ncomp-1 if config["k_comp"] is None else config["k_comp"]
+        self.k_comp = self.ncomp - 1 if config["k_comp"] is None \
+            else config["k_comp"]
+        self.env_type = config["env_type"]
         self.time = 0
         self.ep_length = 30
         self.nstcomp = 30  # What is this?
```

Comment:
self.nstcomp
self.nsthyperp
self.nobs

We maybe need to rename these variables and explicitly state their usefulness.

Reply:
I have added some comments there.